sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileClient.java

```diff
@@ -1121,12 +1178,63 @@ public ShareFileDownloadResponse downloadWithResponse(OutputStream stream, Share
     public ShareFileDownloadResponse downloadWithResponse(OutputStream stream, ShareFileDownloadOptions options,
         Duration timeout, Context context) {
         Objects.requireNonNull(stream, "'stream' cannot be null.");
+        options = options == null ? new ShareFileDownloadOptions() : options;
+        ShareFileRange range = options.getRange() == null ? new ShareFileRange(0) : options.getRange();
```

Comment:
`downloadToFileWithResponse` and `downloadWithResponse` can be changed to use a common method with an additional parameter on how to write the content, but that can be done in a later change. It would just be a case of adding another internal method where it takes all the download concepts (range, HTTP preconditions, lease information, etc) and takes an additional callback function that performs writing the chunk to the output location (file, OutputStream, etc [and this would be a good time to add a method for Java 8 channels]).